forcedotcom/salesforcedx-vscode-automation-tests
# runApexTests.test.ts runs successfully on remote

### How to test:

(1) Go to the [End to End Tests Workflow](https://github.com/forcedotcom/salesforcedx-vscode/actions/workflows/e2e.yml) in salesforcedx-vscode repo
(2) Paste the PR's branch (daphne/runApexTests-on-remote) in the second field of the Run workflow dropdown (Set the branch to use for automation tests)
(3) Choose runApexTests.e2e.ts test from the dropdown to select the test to run.

AC:
Test suite should pass.

### Key changes made:

(1) Added logic to trigger refreshment of the APEX TESTS section of the sidebar if tests do not show up.  (Step 'Run all Apex tests via Test Sidebar' in runApexTests.test.ts)

(2) Added a new function notificationIsPresentWithTimeout().  This function searches for the presence of a given notification within a specified amount of time.  (notifications.ts)

(3) Removed all calls to waitForNotificationToGoAway().  Originally, this function is called as a pause after every time a test is run, so that there is enough time for the run to finish before the code checks that the test successfully ran.  With the implementation of the notificationIsPresentWithTimeout() function, it is no longer necessary to call waitForNotificationToGoAway().  (All steps of runApexTests.test.ts)

(4) Added code that clears the text from the Output view.  This is done right before every place where Apex tests are run.  (All steps that run Apex tests in runApexTests.test.ts)

(5) Added additional checks on the output table that shows up in the Output view after tests are run in order to ensure that all tests pass.  (All steps that run Apex tests in runApexTests.test.ts)

### Successful runs:
• [244](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5836982238)
• [245](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5836985021)
• [246](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5836986138)
• [247](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5836987655)

@W-13744002@
